apps/src/lib/tools/jsinterpreter/JSInterpreter.js

```diff
@@ -241,6 +241,27 @@ export default class JSInterpreter {
     }
   }
 
+  /**
+   * Finds all functions within a section of code and returns on object
+   * representing them and their input parameters.
+   *
+   * @param {string} [code] - a string containing the code to be parsed for functions.
+   */
+  getFunctionsAndParams(code) {
+    this.parse({code: code});
```

Comment:
Will this throw away the previously-parsed AST? Trying to understand the side effects of calling this method.